src/util/prolog/prologStringManipulation.ts

```diff
@@ -1,55 +1,94 @@
 import {fetchAllMatchesForAGroup} from "../regExpUtils";
 import builtInPredicates from "./builtInPredicates";
 
-export function removeFileExtension(path: string) {
-	const fetchFileExtension = /(.*)\.[^\.]*/;
-
+/**
+	* @param path - The path to be curated
+	* @return The path without its extension (removing the last dot and everything after)
+	*/
+export function removeFileExtension(path: string): string {
+	const fetchFileExtension = /(.+)\.[^\.]*/;
 	const fetch: string[] | null = fetchFileExtension.exec(path);
 	if (fetch === null) { return ""; }
 	return fetch[1];
 }
 
-export function removeModuleDeclarations(prologProgram: string) {
+/**
+	* Necessary because module declarations break tau-prolog
+	* @param path - The program code to be curated
+	* @return Program code without module declarations except for library calls
+	*/
+export function removeModuleDeclarations(prologProgram: string): string {
 	const fetchModuleDeclaration = /:-( |)module\([^)]*\) *\.[\r\t\f\v ]*\n?/g;
-
 	return prologProgram.replace(fetchModuleDeclaration, "");
 }
 
-function removeFullLineComments(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without prolog comments that span a whole line including the line
+	*/
+function removeFullLineComments(prologProgram: string): string {
 	const fetchFullLineCommentsRegExp = /^[\r\t\f\v ]*%[^\n]*\n/gm;
-
 	return prologProgram.replace(fetchFullLineCommentsRegExp, "");
 }
 
-function removePartialLineComments(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without prolog comments at the end of line
+	*/
+function removePartialLineComments(prologProgram: string): string {
 	const fetchPartialLineCommentsRegExp = /(^[^\n%]+)%[^\n\r]*(\r?\n)/gm;
-
 	// group2 fetches the end-of-line (EOL) standard in the file to fit both: Windows and UNIX
 	// https://www.networkworld.com/article/3107972/windows-vs-unix-those-pesky-line-terminators.html
 	return prologProgram.replace(fetchPartialLineCommentsRegExp, (_, group1, group2) => group1.trimRight() + group2);
 }
 
-export function removeComments(prologProgram: string) {
-
- return removeFullLineComments(removePartialLineComments(prologProgram));
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without prolog comments and the lines of full line comments
+	*/
+export function removeComments(prologProgram: string): string {
+	return removeFullLineComments(removePartialLineComments(prologProgram));
 }
 
-function removeEmptyLines(prologProgram: string) {
+/**
+	* To reduce code
+	* @param path - The program code to be curated
+	* @return Program code without empty lines and lines with only whitespace characters
+	*/
+function removeEmptyLines(prologProgram: string): string {
 	const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
```

Comment:
It's slightly more performant and readable to declare all these regexes once at the top of the file instead of declaring them inside the function; it currently creates a new instance of them every time the function is called, which is unnecessary.

Reply:
I see. They used to be all at the top of the file, but I decided to pull them into the function for readability and since these regular expressions couldn't be reused.

@Nifdee what would you think about declaring them outside of the function above the function signature like:

```
const fetchEmptyLinesRegExp = /\n(?:\s)*\n/gm;
function removeEmptyLines(prologProgram: string): string {
[...]
}
```